### PR TITLE
json-converted-to-xml

### PR DESCRIPTION
**Pull request by qadan**

### CODE REVIEW

**Comment by adam-vessey** [comment]
Depends on https://github.com/discoverygarden/dgi_gsearch_extensions/pull/3

File: foxmlToSolr.xslt [hunk 1]
```diff
@@ -225,6 +226,15 @@
               <xsl:with-param name="content" select="document(concat($PROT, '://', encoder:encode($FEDORAUSER), ':', encoder:encode($FEDORAPASS), '@', $HOST, ':', $PORT, '/fedora/objects/', $PID, '/datastreams/', @ID, '/content'))"/>
             </xsl:apply-templates>
           </xsl:when>
+          <!-- JSON to document objects -->
+          <xsl:when test="@CONTROL_GROUP='M' and foxml:datastreamVersion[last() and @MIMETYPE='application/json']">
+            <xsl:variable name="json">
+              <xsl:value-of select="java:ca.discoverygarden.gsearch_extensions.FedoraUtils.getRawDatastreamDissemination($PID, @ID, concat($PROT, '://', $HOST, ':', $PORT, '/fedora'), $FEDORAUSER, $FEDORAPASS)"/>
```
Comment (adam-vessey) [inline foxmlToSolr.xslt]: The `select` could be moved inline in the variable, couldn't it? Like:
```xsl
<xsl:variable name="json"
  select="java:ca.discoverygarden.gsearch_extensions.FedoraUtils.getRawDatastreamDissemination($PID, @ID, concat($PROT, '://', $HOST, ':', $PORT, '/fedora'), $FEDORAUSER, $FEDORAPASS)"/>
```

Comment (adam-vessey) [reply foxmlToSolr.xslt]: Also... could use the other syntax, if we wanted to inline...:
```xsl
<xsl:when test="[... the current test ...]"
  xmlns:dgi-e="xalan://ca.discoverygarden.gsearch_extensions">
  <xsl:variable name="json" select="dgi-e:FedoraUtils.getRawDatastreamDissemination($PID, @ID, concat($PROT, '://', $HOST, ':', $PORT, '/fedora'), $FEDORAUSER, $FEDORAPASS)"/>
  <xsl:apply-templates select="foxml:datastreamVersion[last()]">
    <xsl:with-param name="content" select="dgi-e:JSONToXML.convertJSONToXML($json)"/>
  </xsl:apply-templates>
</xsl:when>
```
or even:
```xsl
<xsl:when test="[... the current test ...]"
  xmlns:dgi-e="xalan://ca.discoverygarden.gsearch_extensions">
  <xsl:apply-templates select="foxml:datastreamVersion[last()]">
    <xsl:with-param name="content" select="dgi-e:JSONToXML.convertJSONToXML(dgi-e:FedoraUtils.getRawDatastreamDissemination($PID, @ID, concat($PROT, '://', $HOST, ':', $PORT, '/fedora'), $FEDORAUSER, $FEDORAPASS))"/>
  </xsl:apply-templates>
</xsl:when>
```
... shrug?

**Review by adam-vessey — COMMENTED** [review]
Looks good... couple little simplifications possible, but not necessary.

**Review by adam-vessey — APPROVED** [review]

**Review by adam-vessey — CHANGES_REQUESTED** [review]
Needs to be updated with https://github.com/discoverygarden/islandora_transforms/pull/7

**Comment by qadan** [comment]
For the basic solr config, I don't believe we target a specific commit of islandora_transforms, we just specify HEAD, so should be fine for this one?

**Comment by adam-vessey** [comment]
Looks like a particular commit, to me: image

**Comment by qadan** [comment]
screen shot 2017-04-19 at 10 16 26 am

I think that's just meant to be an informative link. The `.gitmodules` file looks thusly

**Comment by adam-vessey** [comment]
But:
```
velociraptor:repos adam$ git clone https://github.com/discoverygarden/basic-solr-config asdf
Cloning into 'asdf'...
remote: Counting objects: 1228, done.
remote: Total 1228 (delta 0), reused 0 (delta 0), pack-reused 1228
Receiving objects: 100% (1228/1228), 356.15 KiB | 491.00 KiB/s, done.
Resolving deltas: 100% (670/670), done.
velociraptor:repos adam$ cd asdf
velociraptor:asdf adam$ git submodule status islandora_transforms
-fc07d83f4004c6fa2e2210f390c925bc7adf216c islandora_transforms
```
... I'm not sure where exactly it has this info stored...

**Comment by adam-vessey** [comment]
Similarly:
```
velociraptor:repos adam$ git clone https://github.com/discoverygarden/basic-solr-config qwer --recursive
Cloning into 'qwer'...
remote: Counting objects: 1228, done.
remote: Total 1228 (delta 0), reused 0 (delta 0), pack-reused 1228
Receiving objects: 100% (1228/1228), 356.15 KiB | 446.00 KiB/s, done.
Resolving deltas: 100% (670/670), done.
Submodule 'islandora_transforms' (https://github.com/discoverygarden/islandora_transforms.git) registered for path 'islandora_transforms'
Cloning into '/Users/adam/repos/qwer/islandora_transforms'...
remote: Counting objects: 998, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 998 (delta 0), reused 0 (delta 0), pack-reused 994
Receiving objects: 100% (998/998), 318.81 KiB | 570.00 KiB/s, done.
Resolving deltas: 100% (545/545), done.
Submodule path 'islandora_transforms': checked out 'fc07d83f4004c6fa2e2210f390c925bc7adf216c'
velociraptor:repos adam$
```

**Comment by qadan** [comment]
figured it out

submodules is goofy

**Review by adam-vessey — APPROVED** [review]

**Pull request merged**